workspace_integration_test.go

```diff
@@ -1211,6 +1218,134 @@ func TestWorkspacesDeleteByID(t *testing.T) {
 	})
 }
 
+func TestCanForceDeletePermission(t *testing.T) {
+	skipIfNotCINode(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	t.Cleanup(orgTestCleanup)
+
+	wTest, wCleanup := createWorkspace(t, client, orgTest)
+	t.Cleanup(wCleanup)
+
+	t.Run("workspace permission set includes can-force-delete", func(t *testing.T) {
+		w, err := client.Workspaces.ReadByID(ctx, wTest.ID)
+		require.NoError(t, err)
+		assert.Equal(t, wTest, w)
+		assert.NotNil(t, w.Permissions.CanForceDelete)
+		assert.True(t, *w.Permissions.CanForceDelete)
+	})
+}
+
+func TestWorkspacesSafeDelete(t *testing.T) {
+	skipIfNotCINode(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	t.Cleanup(orgTestCleanup)
+
+	wTest, wCleanup := createWorkspace(t, client, orgTest)
+	t.Cleanup(wCleanup)
+
+	t.Run("with valid options", func(t *testing.T) {
+		err := client.Workspaces.SafeDelete(ctx, orgTest.Name, wTest.Name)
+		require.NoError(t, err)
+
+		// Try loading the workspace - it should fail.
+		_, err = client.Workspaces.Read(ctx, orgTest.Name, wTest.Name)
+		assert.Equal(t, ErrResourceNotFound, err)
+	})
+
+	t.Run("when organization is invalid", func(t *testing.T) {
+		err := client.Workspaces.SafeDelete(ctx, badIdentifier, wTest.Name)
+		assert.EqualError(t, err, ErrInvalidOrg.Error())
+	})
+
+	t.Run("when workspace is invalid", func(t *testing.T) {
+		err := client.Workspaces.SafeDelete(ctx, orgTest.Name, badIdentifier)
+		assert.EqualError(t, err, ErrInvalidWorkspaceValue.Error())
+	})
+
+	t.Run("when workspace is locked", func(t *testing.T) {
+		wTest, workspaceCleanup := createWorkspace(t, client, orgTest)
+		t.Cleanup(workspaceCleanup)
+		w, err := client.Workspaces.Lock(ctx, wTest.ID, WorkspaceLockOptions{})
+		assert.NoError(t, err)
```

Comment:
missed one `require.NoError(t, err)`, I'd even say we should also `require.True(t, w.Locked)` since the rest of your test depends on this being true. If this fails, there's no point in continuing to run the rest of the test.